lib/rubocop/cop/naming/block_forwarding.rb

```diff
@@ -47,11 +47,17 @@ class BlockForwarding < Base
 
         MSG = 'Use %<style>s block forwarding.'
 
+        # @!method block_assignment?(node, var_name)
+        def_node_search :block_assignment?, <<~PATTERN
+          (lvasgn %1)
+        PATTERN
```

Comment:
Perhaps the following change to the production code on L110 instead of this matcher will cover the issue. 
How about this suggestion?

```diff
- node.body.each_descendant(:lvar).any? do |lvar|
+ node.body.each_descendant(:lvar, :lvasgn).any? do |lvar|
```